clang/lib/Sema/SemaTemplateInstantiateDecl.cpp

```diff
@@ -775,6 +775,12 @@ void Sema::InstantiateAttrs(const MultiLevelTemplateArgumentList &TemplateArgs,
           *this, TemplateArgs, SYCLIntelMaxGlobalWorkDim, New);
       continue;
     }
+    if (const auto *SYCLIntelNoGlobalWorkOffset =
+            dyn_cast<SYCLIntelNoGlobalWorkOffsetAttr>(TmplAttr)) {
+      instantiateIntelSYCLFunctionAttr<SYCLIntelNoGlobalWorkOffsetAttr>(
```

Comment:
And somewhere around here (either within this helper function, if appropriate, or here in `InstantiateAttrs()`) we should be checking that the instantiated expression is an integer constant expression.

Reply:
We are handling this inside Sema.h file.